src/traces/sunburst/plot.js

```diff
@@ -0,0 +1,806 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var d3 = require('d3');
+var d3Hierarchy = require('d3-hierarchy');
+
+var Registry = require('../../registry');
+var Fx = require('../../components/fx');
+var Color = require('../../components/color');
+var Drawing = require('../../components/drawing');
+var Lib = require('../../lib');
+var Events = require('../../lib/events');
+var svgTextUtils = require('../../lib/svg_text_utils');
+var setCursor = require('../../lib/setcursor');
+var appendArrayPointValue = require('../../components/fx/helpers').appendArrayPointValue;
+
+var transformInsideText = require('../pie/plot').transformInsideText;
+var formatPieValue = require('../pie/helpers').formatPieValue;
+var styleOne = require('./style').styleOne;
+
+var CLICK_TRANSITION_TIME = 1000;
+var CLICK_TRANSITION_EASING = 'linear'; // TODO or something else?
```

Comment:
> agree on click transitions easing and duration

I vote for `linear` easing as we currently have. 

As for the transition time, 1000ms is nice to showcase the animation but it is too long considering it prevents from clicking again. This means that after zooming 4 levels deep with one click, it then takes 4 clicks times 1000ms to get back to the root again. This is too long. I think we shouldn't prevent clicking for more than 500 ms.

Therefore, I personally like anything between 250 and 500ms. We have time to see the animation but it doesn't block clicking.

@etpinard, maybe in the future we could this configurable via an attribute?

Reply:
Hmm. 500ms is fast, too fast for my liking:

![Peek 2019-03-28 09-47](https://user-images.githubusercontent.com/6675409/55162612-9029ee00-513e-11e9-852b-38ceaf5d4e86.gif)

Can we settle on 750ms?



> we shouldn't prevent clicking for more than 500 ms.

You're right here. It would be nice to not have to prevent clicks during transitions. We'll keep that in mind for the next iteration.

Reply:
> You're right here. It would be nice to not have to prevent clicks during transitions. We'll keep that in mind for the next iteration.

If you don't have to prevent clicks during transitions, I could even agree with going with 1000ms! I love the animations :smile: In the meantime, if you prefer 750ms, let's go with that!

In a future iteration, I'd be curious to see it makes sense to make the animation time proportional to the number of levels you're zooming in. The deeper you zoom, the more things move around and maybe longer animations would be beneficial for those cases while zooming only one level would be faster :thinking:

Reply:
Lowered to 750ms in https://github.com/plotly/plotly.js/pull/3594/commits/b59be1d732b8412dfda8af60e114cb1c4050806b